### ISSUE TITLE
Create missing basic elements with "Insert"

### ISSUE THREAD

**Issue by anewuser** [pipes-digital/pipes#110]
See this example: https://www.pipes.digital/pipe/E9D7p4qj

If you set the `insert` block to target `/rss/channel/item/description`, it successfully replaces the existing description.

However, if you target `/rss/channel/item/title`, nothing happens, since the post has no `title` tag yet.

**Comment by onli** [comment]
That would be linked to https://github.com/pipes-digital/pipes/issues/92.

Simple problem: I don't know how to do this properly. Nokogiri does not have an API to create an XML element at a random position via an xpath. At least I did not see a proper solution.

So the code would devolve to something like: Split the xpath at `/`, always append one element more together until the xpath does not find the element, then create the children for all subsequent parts of the xpath. That felt so brittle that it gave me pause.

Do you see a better solution?

**Comment by anewuser** [comment]
Doesn't [.add_next_sibling](https://nokogiri.org/tutorials/modifying_an_html_xml_document.html#creating-new-nodes) work with XPath results?

If it does, what about following this logic:

    if xpath(target) exists:
        execute current code
    elseif xpath(target) matches '^\/rss\/channel\/item\/(title|description|content:encoded)$' 
       xpath('/rss/channel/item/link').add_next_sibling <replace>content</replace>
       
I think it's a good idea to hardcode the accepted tags to prevent abuse or random errors.

**Comment by onli** [comment]
> Doesn't [.add_next_sibling](https://nokogiri.org/tutorials/modifying_an_html_xml_document.html#creating-new-nodes) work with XPath results?

I don't think so. add_next_sibling in general is cumbersome, and it does not work with xpath for me at all:

```
onli@fallout:~/Dropbox-decrypted/pipes$ irb
irb(main):001:0> require 'nokogiri'
=> true
irb(main):002:0> doc = Nokogiri::XML('<xml><a></a></xml>')
=> 
#(Document:0x742c {                                                   
...                                                                   
irb(main):003:0> doc.add_next_sibling('<b></b>')
/home/onli/.gem/ruby/3.1.0/gems/nokogiri-1.13.8-x86_64-linux/lib/nokogiri/xml/node.rb:219:in `add_next_sibling': undefined local variable or method `parent' for #<Nokogiri::XML::Document:0x742c name="document" children=[#<Nokogiri::XML::Element:0x7580 name="xml" children=[#<Nokogiri::XML::Element:0x76d4 name="a">]>]> (NameError)
                                                                      
          "A document may not have multiple root nodes." if parent&.document? && !(node_or_tags.comment? || node_or_tags.processing_instruction?)                                             
                                                            ^^^^^^    
Did you mean?  parent=                                                
               print                                                  
	from (irb):3:in `<main>'                     
	from /usr/lib64/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /usr/sbin/irb:25:in `load'              
	from /usr/sbin/irb:25:in `<main>'            
irb(main):004:0> a = doc.at_css('a')
=> #(Element:0x76d4 { name = "a" })
irb(main):005:0> a.add_next_sibling('<b></b>')
=> [#<Nokogiri::XML::Element:0x1bea4 name="b">]
irb(main):006:0> doc.to_s
=> "<?xml version=\"1.0\"?>\n<xml>\n  <a/>\n  <b/>\n</xml>\n"
irb(main):007:0> b = doc.at_css('b')
=> #(Element:0x1bea4 { name = "b" })
irb(main):008:0> b.add_next_sibling('/c/d')
=> [#<Nokogiri::XML::Text:0x2d5f0 "/c/d">]
irb(main):009:0> doc.to_s
=> "<?xml version=\"1.0\"?>\n<xml><a/><b/>/c/d</xml>\n"
```

And there is no add_at_xpath. The code would have to get the target element(s!) and then add the siblings manually.

**Comment by onli** [comment]
Note: I just deleted a comment where I understood your suggestion wrong. Your idea to limit it to specific items could work. There is no reason not to have them in our output feed anyway. Could be automated whenever a RSS feed is generated. I dislike that it's not a general solution, but it should solve the reported usecase.

**Comment by anewuser** [comment]
>Could be automated whenever a RSS feed is generated.

This would actually be my next suggestion. If the `feed` block adds the missing basic elements to all items automatically, then we'll be able to keep all items in a problematic feed without having to recreate or truncate the whole feed.

What about using the date as a placeholder title in case there's no title? With #71, it'd be even easier to customize each title afterwards.
